### ISSUE TITLE
Idefics support in BetterTransformer

### ISSUE THREAD

**Issue by Patrykkrawczak** [huggingface/optimum#1336]
### Feature request

add idefics to supported models for BetterTransformer

### Motivation

It is not possible currently to use idefics model efficiently (as its too big for most GPUs)

### Your contribution

i can contribute as best as i can to the implementation

**Comment by fxmarty** [comment]
Hi @Patrykkrawczak, thank you! IDEFICS already uses SDPA in transformers. BetterTransformer integration for IDEFICS would basically replacing https://github.com/huggingface/transformers/blob/bfb1895e3346cb8a2bf2560c75d45e70edf46a47/src/transformers/models/idefics/modeling_idefics.py#L692 by a call to SDPA that uses the attention_mask only for batch size > 1 during training, as SDPA can not dispatch to memory-efficient attention / flash attention when an attention mask is passed (as of torch 2.0.1).

This would look for example roughly like this: https://github.com/huggingface/optimum/blob/659cf0267706e48c711e7c503d3057b7538efe0e/optimum/bettertransformer/models/attention.py#L640-L660

**Comment by fxmarty** [comment]
Hi @Patrykkrawczak for reference: https://huggingface.co/docs/transformers/perf_infer_gpu_one#flashattention-and-memory-efficient-attention-through-pytorchs-scaleddotproductattention
